### ISSUE TITLE
Internal Error: Do not pass posix paths to Windows APIs when `bun add react-dom/server`

### ISSUE THREAD

**Issue by WhatNot911** [oven-sh/bun#10397]
### What version of Bun is running?

1.1.4+fbe2fe0c3

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64

### What steps can reproduce the bug?

1. bun init
2. cd to created project
3. bun add react-dom/server

### What is the expected behavior?

I don't really know what needs to happen, I suppose bun had to add react-dom/server package to node_modules so I can use it like in this tutorial? 
[https://bun.sh/guides/ecosystem/ssr-react](https://bun.sh/guides/ecosystem/ssr-react)

Maybe if said package doesn't exist it should say something different?

### What do you see instead?

```
uh-oh: Internal Error: Do not pass posix paths to Windows APIs, was given '\.bun-tag'. Please open an issue on GitHub with a reproduction.
bun will crash now 😭😭😭

----- bun meta -----
Bun v1.1.4 (fbe2fe0c) Windows x64
AddCommand: dotenv extracted_packages
Elapsed: 4329ms | User: 187ms | Sys: 484ms
RSS: 88.85MB | Peak: 112.57MB | Commit: 0.14GB | Faults: 28413
----- bun meta -----

0   00007FF62F022588
1   ???
2   ???
3   ???
4   ???
5   ???
6   ???
7   ???
8   ???
9   ???
10  ???
11  ???
12  ???
13  ???
14  ???
15  ???
16  ???
17  ???

Search GitHub issues https://bun.sh/issues or join in #windows channel in https://bun.sh/discord

thread 26140 panic: Internal Error: Do not pass posix paths to Windows APIs, was given '\.bun-tag'. Please open an issue on GitHub with a reproduction.
???:?:?: 0x7ff62e20d2f3 in ??? (bun.exe)
???:?:?: 0x7ff62df58a58 in ??? (bun.exe)
???:?:?: 0x7ff62dd8b946 in ??? (bun.exe)
???:?:?: 0x7ff62dfad9f9 in ??? (bun.exe)
???:?:?: 0x7ff62dbdb042 in ??? (bun.exe)
???:?:?: 0x7ff62e1e15a8 in ??? (bun.exe)
???:?:?: 0x7ff62db95b7b in ??? (bun.exe)
???:?:?: 0x7ff62db93a2b in ??? (bun.exe)
???:?:?: 0x7ff62db89669 in ??? (bun.exe)
???:?:?: 0x7ff62db85e97 in ??? (bun.exe)
???:?:?: 0x7ff62db998e3 in ??? (bun.exe)
???:?:?: 0x7ff62dc3fa65 in ??? (bun.exe)
???:?:?: 0x7ff62d961603 in ??? (bun.exe)
???:?:?: 0x7ff630524c63 in ??? (bun.exe)
???:?:?: 0x7ff9a60d7343 in ??? (KERNEL32.DLL)
???:?:?: 0x7ff9a77a26b0 in ??? (ntdll.dll)
```

## Stacktrace

**edited**

Bun v1.1.5 ([`1a4bad8`](https://github.com/oven-sh/bun/tree/1a4bad80b7a51a79e0a1570a053e40eb33af8b94)) on windows x86_64 [AddCommand]

**panic**: Internal Error: Do not pass posix paths to Windows APIs, was given `'\.bun-tag'`. Please open an issue on GitHub with a reproduction.

- [`js_ast.zig:3040`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/js_ast.zig#L3040): `assert`
- [`debug.zig:348`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/deps/zig/lib/std/debug.zig#L348): `panic`
- [`bun.zig:3143`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/bun.zig#L3143): `unsafeAssert`
- [`sys.zig:826`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/sys.zig#L826): `openFileAtWindowsNtPath`
- [`sys.zig:969`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/sys.zig#L969): `openFileAtWindowsT`
- [`install.zig:972`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/install/install.zig#L972): `verifyGitResolution`
- [`install.zig:9241`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/install/install.zig#L9241): `installPackageWithNameAndResolution`
- [`install.zig:10089`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/install/install.zig#L10089): `installPackages`
- [`install.zig:10776`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/install/install.zig#L10776): `installWithManager`
- [`install.zig:8487`](https://github.com/oven-sh/bun/blob/1a4bad80b7a51a79e0a1570a053e40eb33af8b94/src/install/install.zig#L8487): `updatePackageJSONAndInstallWithManagerWithUpdates`

Features:

**Comment by brianferri** [comment]
Just got the same error in a different context. Weird thing is that it was working just yesterday
On: `Microsoft Windows NT 10.0.22631.0 x64`
This does not happen on WSL: `Linux 5.10.102.1-microsoft-standard-WSL2 x86_64 x86_64`
```sh
bun test v1.1.4 (fbe2fe0c)

tests\middleware.test.ts:
✓ MIDDLEWARE > runs middleware on server scope [22.68ms]
✓ MIDDLEWARE > runs middleware on route scope [5.39ms]
✓ MIDDLEWARE > runs multiple middlewares on route scope [4.98ms]
✓ MIDDLEWARE > runs middleware on dynamic route scope [5.64ms]
✓ MIDDLEWARE > runs middleware on dynamic route on server scope [5.54ms]
✓ MIDDLEWARE > runs middleware with response return [5.94ms]
✓ MIDDLEWARE > runs middleware with multiple response return [4.37ms]
✓ MIDDLEWARE > runs middleware with multiple response return on server scope [5.42ms]
✓ MIDDLEWARE > runs middleware with multiple response return on dynamic route [6.46ms]
✓ MIDDLEWARE > runs middleware with multiple response return on dynamic route on server and route scope [3.34ms]

uh-oh: Internal Error: Do not pass posix paths to Windows APIs, was given '\'. Please open an issue on GitHub with a reproduction.
bun will crash now 😭😭😭

----- bun meta -----
Bun v1.1.4 (fbe2fe0c) Windows x64
TestCommand: tsconfig(2) http_server(10)
Elapsed: 174ms | User: 15ms | Sys: 15ms
RSS: 122.37MB | Peak: 122.37MB | Commit: 0.17GB | Faults: 30089
----- bun meta -----

0   00007FF796C32588
1   ???
2   ???
3   ???
4   ???
5   ???
6   ???
7   ???
8   ???
9   ???
10  ???
11  ???

Search GitHub issues https://bun.sh/issues or join in #windows channel in https://bun.sh/discord

thread 4496 panic: Internal Error: Do not pass posix paths to Windows APIs, was given '\'. Please open an issue on GitHub with a reproduction.
???:?:?: 0x7ff795e1d2f3 in ??? (bun.exe)
???:?:?: 0x7ff795b68a58 in ??? (bun.exe)
???:?:?: 0x7ff79599b946 in ??? (bun.exe)
???:?:?: 0x7ff795aff7c0 in ??? (bun.exe)
???:?:?: 0x7ff7955be727 in ??? (bun.exe)
???:?:?: 0x7ff7955bcf23 in ??? (bun.exe)
???:?:?: 0x7ff7955c0822 in ??? (bun.exe)
???:?:?: 0x7ff795d09a1c in ??? (bun.exe)
???:?:?: 0x7ff797dfe9aa in ??? (bun.exe)
???:?:?: 0x1870a9615cb in ??? (???)
???:?:?: 0x573bd8498f in ??? (???)
???:?:?: 0x7ff79811102d in ??? (bun.exe)
error: script "release:dry" exited with code 3
```
This is happening during `bun test` on a test file which does not contain any `\` whatsoever

Repro by cloning the master branch from https://github.com/DotEnv-it/futen (https://git.dpaste.org/BioCla/futen) and running `bun release:dry`

**Comment by Jarred-Sumner** [comment]
@BioCla please open a new issue for that. The error is the same, but the cause is different.

**Comment by WhatNot911** [comment]
As of 1.1.7 the error log somewhat changed but ultimately stays the same

```
bun add react-dom/server
bun add v1.1.7 (b0b7db5c)
============================================================
Bun v1.1.7 (b0b7db5c) Windows x64
Args: "bun", "add", "react-dom/server"
Features: extracted_packages
Elapsed: 3337ms | User: 93ms | Sys: 625ms
RSS: 121.54MB | Peak: 121.54MB | Commit: 0.18GB | Faults: 35602

panic(main thread): Internal Error: Do not pass posix paths to Windows APIs, was given '\.bun-tag'. Please open an issue on GitHub with a reproduction.
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.7/wI1b0b7db5AggBk9voQ+hhlLup62Hgml5Lmt29H8lq+Pi96mE4trmE6u9jEqsjjEA0eNoFwTEOwjAMBdCr/K0LzQHYkKigWzcWFpdaraXKjmyHcHzemzXZlU5M7uZX3A1qiUoRqBbyQ6U8Aml4iW7WA7dljgs6BXb5smJ4l7XpmLQPBcvJFAyrrCCFRDSGKR6Sz7aiSx4gOFe3rX1STMsf3qUuHA
 ```
 
 Just in case anyone is curious of what I'm trying to do adding react-dom/server, here is the reason:
 I'm trying to do SSR with react. I need to use function renderToReadableStream, but I can't:
 ```SyntaxError: Export named 'renderToReadableStream' not found in module '\\app\node_modules\react-dom\server.node.js'.```
I'm following this tutorial (JIC) : https://bun.sh/guides/ecosystem/ssr-react
I think it could work on Linux machines, but at this point can't verify it. Should I open a different issue for this at all by the way?

**Comment by Jarred-Sumner** [comment]
@WhatNot911 `bun add react-dom/server` attempts to run the equivalent of `bun add ./react-dom/server` which has a bug where the `/` isn't being normalized to `\\` on Windows leading to the assertion failure in this thread.

The fix for the specific issue you're experiencing is to change the import path you're using to `react-dom/server.browser` instead of `react-dom/server`, or to upgrade to React 19 `bun add react@next react-dom@next`.

Related: 
- https://github.com/facebook/react/pull/28962 
- https://github.com/facebook/react/issues/28941

**Comment by WhatNot911** [comment]
Thank you very much! ❤️

**Comment by dylan-conway** [comment]
Fixed in #11644, and will be included in v1.1.13. Please reopen or open a new issue if you continue to have issues related to this.
